### PR TITLE
undefined varialbe fail test

### LINKED ISSUE #51: Undefined variable in FIND
need to throw exception for undefined variable #50 
May need to fix this after we have our dependency

### PR DESCRIPTION
**Pull request by rxie25**
fixes #51; Semnat.ml needs to throw undefined variable in this case

### CODE REVIEW

File: src/tests/fail-undefined-varaible.skip.eq [hunk 1]
```diff
@@ -0,0 +1,11 @@
+TownCtx= {
+  people = {17;}
+}
+
+FarmCtx = {
```
Comment (nnhoang) [inline src/tests/fail-undefined-varaible.skip.eq]: what's the `FarmCtx` for?

**Comment by jzacsh** [comment]
I think looks good, once you address Nam's comment

File: src/tests/fail-undefined-varaible.skip.eq (hunk 1 above)
Comment (rxie25) [inline src/tests/fail-undefined-varaible.skip.eq]: I was for another test. I forget to remove it completely. I guess it doesn't test anything in this case. I will delete it.

**Comment by rxie25** [comment]
Since our current master can't pass the test. It's added as skip.

**Comment by jzacsh** [comment]
If you want, you should be able to wait a few minutes and merge in Nam's work and add your own `raise`

**Comment by rxie25** [comment]
ok, I will wait for nam to push the code and add semantic checking and make sure the test passes.

**Comment by rxie25** [comment]
nam already fixed the issue on his branch. I also added a fail test for Tianci's generating separate function for different context. #47

File: src/tests/fail-undefined-variable2.eq [hunk 2]
```diff
@@ -3,7 +3,7 @@ TownCtx= {
 }
```
Comment (jzacsh) [inline src/tests/fail-undefined-variable2.eq]: Can you give this file a better name than "2"?

**Comment by jzacsh** [comment]
just assign back to me when you want me to take another look

**Comment by rxie25** [comment]
I'll rename them.

**Comment by rxie25** [comment]
the skipped test (fail-undef-var-with-def-target.skip.eq) will get resolved once we have a semantic check for the variable IDs in find block .

**Comment by jzacsh** [comment]
Looks good!

**Pull request merged**